scripts/regtest-watcher.js

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env node
+
+const logger = require('pino')()
+const Client = require('bitcoin-core')
+
+const {
+  BITCOIN_URL: host = 'bitcoind-1',
+  BITCOIN_PORT: port = '18443',
+  BITCOIN_USER: username  = "bitcoinrpcuser",
+  BITCOIN_PASSWORD: password = "bitcoinrpcpassword",
+  BITCOIN_NEW_BLOCK_INTERVAL_MS: newBlockInterval = 60000,
+} = process.env
+
+const bitcoindClientA = new Client({ host, port, network: 'regtest', username, password })
+
+const main = async () => {
+  try {
+    logger.info('Checking wallet balance')
+    const balance = await bitcoindClientA.getBalance()
+    logger.info(`Wallet balance: ${balance}`)
```

Comment:
you can pass objects as first argument to Pino: `logger.info({ balance }, 'Wallet balance retrieved')`.

Pino will take care of pretty-printing it etc